components/raftstore/src/store/peer.rs

```diff
@@ -3676,6 +3679,9 @@ where
             // The message is dropped silently, this usually due to leader absence
             // or transferring leader. Both cases can be considered as NotLeader error.
             return Err(Error::NotLeader(self.region_id, None));
+        } else if self.force_leader {
```

Comment:
minor suggestion: else if can be if.  Same for other similar coded.
also maybe it's easier to read and maintain to create a tiny method forward_commit_index to wrap the logic as it appears in multiple places.

Reply:
forwarding commit index is simple enough, no need to extract it.